File: DesignPatterns/src/Java/002-GOF-Patterns/002-StructuralPatterns/003-Facade/DeviceAppDB/src/org/csystem/app/PortInfoRepository.java
package org.csystem.app;

public class PortInfoRepository implements IPortInfoRepository {
    @Override
    public <S extends PortInfo> S save(S s)
    {
        System.out.println("PortInfoRepository.save");
        return null;
    }

    @Override
    public Iterable<PortInfo> findAll()
    {
        System.out.println("PortInfoRepository.findAll");
        return null;
    }

    @Override
    public void deleteById(Long aLong)
    {
        System.out.println("PortInfoRepository.deleteById");
    }
}

File: src/DesignPatterns/src/C++/002-GOFPatterns/002-StructuralPatterns/003-Facade/DeviceAppDatabase/PortInfoRepository.cpp
#include <iostream>
#include "PortInfoRepository.h"

PortInfo &PortInfoRepository::save(PortInfo &t)
{
    std::cout << "PortInfoRepository::save\n";
    return t;
}

std::vector<PortInfo> PortInfoRepository::find_all()
{
    std::cout << "PortInfoRepository::find_all\n";

    return std::vector<PortInfo>();
}

bool PortInfoRepository::exist_by_id(const long &id)
{
    std::cout << "PortInfoRepository::exist_by_id\n";

    return false;
}
